packages/contracts/contracts/CollateralManager.sol

```diff
@@ -0,0 +1,372 @@
+     */
+    function initialize(address _collateralEscrowBeacon, address _tellerV2)
+        external
+        initializer
+    {
+        collateralEscrowBeacon = _collateralEscrowBeacon;
+        tellerV2 = TellerV2(_tellerV2);
+    }
+
+    /**
+     * @notice Checks the validity of a borrower's multiple collateral balances and commits it to a bid.
+     * @param _bidId The id of the associated bid.
+     * @param _collateralInfo Additional information about the collateral assets.
+     * @return validation_ Boolean indicating if the collateral balances were validated.
+     */
+    function commitCollateral(
+        uint256 _bidId,
+        ICollateralEscrowV1.Collateral[] calldata _collateralInfo
+    ) public returns (bool validation_) {
+        address borrower = tellerV2.getLoanBorrower(_bidId);
+        (validation_, ) = checkBalances(borrower, _collateralInfo);
+        if (validation_) {
+            for (uint256 i; i < _collateralInfo.length; i++) {
+                ICollateralEscrowV1.Collateral memory info = _collateralInfo[i];
+                _commitCollateral(_bidId, info);
+            }
+            _isBidCollateralBacked[_bidId] = true;
+        }
+    }
+
+    /**
+     * @notice Checks the validity of a borrower's collateral balance and commits it to a bid.
+     * @param _bidId The id of the associated bid.
+     * @param _collateralInfo Additional information about the collateral asset.
+     * @return validation_ Boolean indicating if the collateral balance was validated.
+     */
+    function commitCollateral(
+        uint256 _bidId,
+        ICollateralEscrowV1.Collateral calldata _collateralInfo
+    ) public returns (bool validation_) {
+        address borrower = tellerV2.getLoanBorrower(_bidId);
+        validation_ = _checkBalance(borrower, _collateralInfo);
+        if (validation_) {
+            _commitCollateral(_bidId, _collateralInfo);
+            _isBidCollateralBacked[_bidId] = true;
+        }
+    }
+
+    function revalidateCollateral(uint256 _bidId)
+        external
+        returns (bool validation_)
+    {
+        ICollateralEscrowV1.Collateral[]
+            memory collateralInfos = getCollateralInfo(_bidId);
+        address borrower = tellerV2.getLoanBorrower(_bidId);
+        (validation_, ) = _checkBalances(borrower, collateralInfos, true);
+    }
+
+    function checkBalances(
+        address _borrowerAddress,
+        ICollateralEscrowV1.Collateral[] calldata _collateralInfo
+    ) public returns (bool validated_, bool[] memory checks_) {
+        return _checkBalances(_borrowerAddress, _collateralInfo, false);
+    }
+
+    /**
+     * @notice Deploys a new collateral escrow and deposits collateral.
+     * @param _bidId The associated bidId of the collateral escrow.
+     */
+    function deployAndDeposit(uint256 _bidId) external {
+        if (_isBidCollateralBacked[_bidId]) {
+            require(_msgSender() == address(tellerV2), 'Sender not authorized');
```

Comment:
this could be a modifier named 'onlyTellerV2'  to make this easier to read/cleaner